packages/webamp-modern-2/src/skin/makiClasses/Movable.ts

```diff
@@ -0,0 +1,203 @@
+      case "topright":
+        this._div.style.cursor = "ne-resize";
+        this._resizable = TOP | RIGHT;
+        break;
+      case "bottomleft":
+        this._div.style.cursor = "sw-resize";
+        this._resizable = BOTTOM | LEFT;
+        break;
+      case "bottomright":
+        this._div.style.cursor = "se-resize";
+        this._resizable = BOTTOM | RIGHT;
+        break;
+      default:
+        // this._div.style.removeProperty('cursor');
+        this._resizable = 0;
+    }
+    if (this._movable) {
+      // this._div.style.cursor = "move";
+      this._div.style.removeProperty("cursor");
+      this._resizable = MOVE;
+      this._registerMovingEvents();
+
+      // } else if (this._resizable == 0) {
+    } else if (this._div.style.cursor == "none") {
+      this._div.style.removeProperty("cursor");
+    } else {
+      //   this._div.style.pointerEvents = "auto";
+      this._registerResizingEvents();
+    }
+  }
+
+  _registerResizingEvents() {
+    if (this._resizingEventsRegisterd) {
+      return;
+    }
+    this._resizingEventsRegisterd = true;
+    this._div.addEventListener("mousedown", this._handleResizing.bind(this));
+  }
+
+  _unregisterResizingEvents() {
+    if (this._resizingEventsRegisterd) {
+      this._div.removeEventListener("mousedown", this._handleResizing);
+      this._resizingEventsRegisterd = false;
+    }
+  }
+
+  _handleResizing(downEvent: MouseEvent) {
+    downEvent.stopPropagation();
+    if (downEvent.button != 0) return; // only care LeftButton
+    const layout = this.getparentlayout() as Layout;
+    layout.setResizing("constraint", this._resizable, 0);
+    layout.setResizing("start", 0, 0);
+    layout.setResizing(
+      this._div.style.getPropertyValue("cursor"),
+      CURSOR,
+      CURSOR
+    );
+    const startX = downEvent.clientX;
+    const startY = downEvent.clientY;
+
+    const handleMove = (moveEvent: MouseEvent) => {
+      const newMouseX = moveEvent.clientX;
+      const newMouseY = moveEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setResizing("move", deltaX, deltaY);
+    };
+
+    const handleMouseUp = (upEvent: MouseEvent) => {
+      upEvent.stopPropagation();
+      if (upEvent.button != 0) return; // only care LeftButton
+      document.removeEventListener("mousemove", handleMove);
+      document.removeEventListener("mouseup", handleMouseUp);
+      const newMouseX = upEvent.clientX;
+      const newMouseY = upEvent.clientY;
+      const deltaY = newMouseY - startY;
+      const deltaX = newMouseX - startX;
+      layout.setResizing("final", deltaX, deltaY);
+    };
+    document.addEventListener("mousemove", handleMove);
+    document.addEventListener("mouseup", handleMouseUp);
+  }
+
+  _registerMovingEvents() {
+    if (this._movingEventsRegisterd) {
+      return;
+    }
+    this._movingEventsRegisterd = true;
+    this._div.addEventListener("mousedown", this._handleMoving.bind(this));
```

Comment:
See above comment about event listeners and `.bind`

Reply:
done.